cbits/hs_resolv.h

```diff
@@ -123,6 +131,11 @@ hs_res_query(void *s, const char *dname, int class, int type, unsigned char *ans
   return res_query(dname, class, type, answer, anslen);
 }
 
+inline static void
+hs_res_nclose(struct __res_state *s)
+{
```

Comment:
Why don't we use res_close here?

Reply:
Note that `res_init()`/`res_close()` is a very old interface: the `res_ninit()`/`res_nclose()` appeared in *BIND 8.2* back in 1990s to replace the former. This means that now, in 2020s, the newer interface will most probably always compile in place of those older function (I cannot even find mentioning of `res_close()` in my Fedora 36 *man res_init*).

More important is that the wrapper `hs_res_close()` gets called after `res_ninit()` and `res_nquery()` as it must be called after those functions in order to cleanup resources as *man res_init* suggests. Note that `res_nclose()` accepts an object that will be freed while `res_close()` does not accept arguments as it frees the *global* resource allocated by `res_init()` and is not expected to be called after each `res_query()`!

Reply:
That is reasonable, thanks for the clarification. Overall it looks good for me. It can be considered as a very useful change that prevents resource (memory) leaks and should not affect the overall performance. 

@andreasabel , I would accept it.